advanced/Exercise_7.md

```diff
@@ -1,154 +1,97 @@
 
-```plaintext
-cp -r conference_planning conference_submodule
-```
-
-I have already created a repository on Github that we will use as a submodule. This repository is called `posters` and will contain the poster schedule for our conference.
-
-Navigate to the repository, found [here](https://github.com/kosterried/posters), and make a fork of it using the Github user interface.
-
-Let's add this fork as a submodule to our conference planning repository. You will need to copy the SSH address of your fork and paste it into the `git submodule add` command below.
+Open a terminal and navigate to the `conference_planning` repository on your local machine.
 
-```plaintext
-cd conference_submodule
-git submodule add YOUR_FORK_ADDRESS
-```
-
-Note that Github requires an SSH key in order to push content to repositories. If you have not already set up an SSH key in your Github account, it is easy to do and you can find instructions [here](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/adding-a-new-ssh-key-to-your-github-account).
+## Implement Pre-Commit Hook for Trailing Whitespace <a name="whitespace"></a>
 
-Check the status of the repository.
+Create a new file named `pre-commit` in the `.git/hooks` directory. Make it executable.
 
-```plaintext
-git status
+```sh
+touch .git/hooks/pre-commit
+chmod +x .git/hooks/pre-commit
 ```
 
-The output shows that there are some changes to the repository, and they are already staged for commit. We should commit them to finalize the adding of the submodule.
-
-```plaintext
-git commit -m "Add the posters submodule"
-```
-
-Note that the `.gitmodules` file has been added, which contains the path of the submodule in the host repository and the address where the submodule is hosted which is the Github URL in our case. It could also be another git web host, a file path, or anywhere else a repository is hosted.
-
-Check the status of the submodule.
-
-```plaintext
-git submodule status
-```
-
-The output shows us the SHA of the commit that the submodule is currently pointing to.  
-
-## Examine how git deals with changes to both repositories <a name="examine"></a>
-
-Let's make some changes in both the host repository and the submodule to understand how git deals with submodules.  
+Open the `pre-commit` file in a text editor and add the following script:
 
-Start by making a change in the host repository. Let's add a lunch break to the schedule on day 1. The following gives examples using the `sed` command line tool, which were tested on Linux but may not work on other platforms. You can also simply open the file in a file editor to make the change.
+```bash
+#!/bin/bash
 
+# This is the pre-commit hook to prevent committing files with trailing whitespace.
 
-```plaintext
-sed -i '/^11:00/a 12:00-13:00: Lunch break' schedule_day1
-```
-
-Check the status of the repository.
+# Redirect output to stderr.
+exec 1>&2
 
-```plaintext
-git status
+# Check for trailing whitespace in staged files.
```

Comment:
I don't know what "trailing whitespace" is, so I'm sure there are others who also don't know... A sentence about that would be good.

Reply:
Solved in [9eb535f](https://github.com/C2SM/git-course/pull/83/commits/9eb535fb66e9256ff16b14ddb30ad04178322ca1).